### PR TITLE
roachtest: make roachtest recognize 21.2

### PR DESCRIPTION
**Pull request by celiala**
Shortly after [Creating a release branch](https://cockroachlabs.atlassian.net/wiki/spaces/ENG/pages/187859111/Creating+a+release+branch) for release-21.2, we'll want to merge PRs as instructed by the [New major version checklist](https://cockroachlabs.atlassian.net/wiki/spaces/ENG/pages/1522270228/New+major+version+checklist).

This PR is for Step 2c of the [New major version checklist](https://cockroachlabs.atlassian.net/wiki/spaces/ENG/pages/1522270228/New+major+version+checklist), where we update the ORM tests. Specifically, adding block/ignore lists for the `vNEXT` major version for these files:
```
ls pkg/cmd/roachtest/tests/*_blocklist.go
```

This PR is one of the tasks detailed in #70751, which tracks all the steps relevant to creating a release branch for `${vBRANCH_CUT}` and preparing master for the `${vNEXT}` major version.

Release justification: Non-production code change.
Release note: None

### CODE REVIEW

File: pkg/cmd/roachtest/tests/activerecord_blocklist.go [hunk 1]
```diff
@@ -27,12 +28,17 @@ var activeRecordBlocklists = blocklistsForVersion{
...
 // in the test log.
+var activeRecordBlockList22_1 = blocklist{}
+
 var activeRecordBlockList21_2 = blocklist{}
 
 var activeRecordBlockList21_1 = blocklist{}
 
 var activeRecordBlockList20_2 = blocklist{}
 
+// TODO(celia) -- what should be in blocklist{}, if anything?
```
Comment (celiala) [inline pkg/cmd/roachtest/tests/activerecord_blocklist.go:39]: Maybe @rafiss ?

File: pkg/cmd/roachtest/tests/django_blocklist.go [hunk 2]
```diff
@@ -165,15 +165,21 @@ var djangoBlocklists = blocklistsForVersion{
...
 // Maintain that this list is alphabetized.
+var djangoBlocklist22_1 = djangoBlocklist21_2
+
 var djangoBlocklist21_2 = djangoBlocklist21_1
 
 var djangoBlocklist21_1 = djangoBlocklist20_2
 
 var djangoBlocklist20_2 = blocklist{}
 
+// TODO(celia) -- what should be in blocklist{}, if anything?
```
Comment (celiala) [inline pkg/cmd/roachtest/tests/django_blocklist.go:180]: Maybe @RichardJCai ?

File: pkg/cmd/roachtest/tests/hibernate_blocklist.go [hunk 3]
```diff
@@ -14,20 +14,27 @@ var hibernateBlocklists = blocklistsForVersion{
...
 // Please keep these lists alphabetized for easy diffing.
 // After a failed run, an updated version of this blocklist should be available
 // in the test log.
+var hibernateSpatialBlockList22_1 = blocklist{}
+
 var hibernateSpatialBlockList21_2 = blocklist{}
 
 var hibernateSpatialBlockList21_1 = blocklist{}
 
+// TODO(celia) -- what should be in blocklist{}, if anything?
```
Comment (celiala) [inline pkg/cmd/roachtest/tests/hibernate_blocklist.go:35]: @rafiss ?

File: pkg/cmd/roachtest/tests/pgjdbc_blocklist.go [hunk 4]
```diff
@@ -14,11 +14,15 @@ var pgjdbcBlocklists = blocklistsForVersion{
 	{"v20.2", "pgjdbcBlockList20_2", pgjdbcBlockList20_2, "pgjdbcIgnoreList20_2", pgjdbcIgnoreList20_2},
 	{"v21.1", "pgjdbcBlockList21_1", pgjdbcBlockList21_1, "pgjdbcIgnoreList21_1", pgjdbcIgnoreList21_1},
 	{"v21.2", "pgjdbcBlockList21_2", pgjdbcBlockList21_2, "pgjdbcIgnoreList21_2", pgjdbcIgnoreList21_2},
+	{"v22.1", "pgjdbcBlockList22_1", pgjdbcBlockList22_1, "pgjdbcIgnoreList22_1", pgjdbcIgnoreList22_1},
 }
 
 // Please keep these lists alphabetized for easy diffing.
 // After a failed run, an updated version of this blocklist should be available
 // in the test log.
+// TODO(celia) -- what should be in blocklist{}, if anything?
```
Comment (celiala) [inline pkg/cmd/roachtest/tests/pgjdbc_blocklist.go:23]: @rafiss - here too, thx!

File: pkg/cmd/roachtest/tests/psycopg_blocklist.go [hunk 5]
```diff
@@ -27,6 +28,9 @@ var psycopgBlocklists = blocklistsForVersion{
 // Please keep these lists alphabetized for easy diffing.
 // After a failed run, an updated version of this blocklist should be available
 // in the test log.
+// TODO(celia) -- what should be in blocklist{}, if anything?
```
Comment (celiala) [inline pkg/cmd/roachtest/tests/psycopg_blocklist.go:31]: here too @rafiss - thank you :)

File: pkg/cmd/roachtest/tests/tpchvec.go [hunk 6]
```diff
@@ -31,6 +31,9 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/workload/tpch"
 )
 
+// TODO(celia) -- there used to be a toCRDBVersion() here -- confirm that
```
Comment (celiala) [inline pkg/cmd/roachtest/tests/tpchvec.go:34]: Maybe @yuzefovich ?

**Review by yuzefovich — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/69829)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @celiala)

___
*[pkg/cmd/roachtest/tests/tpchvec.go, line 34 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/69829#-Mihq2oXtlqngbFiy37N-r1-34:-Mj0B8o_DFtyEtrvRnDl:bnkf5k8) ([raw file](https://github.com/cockroachdb/cockroach/blob/70567ad7bef1df02f23b916a9d1e776ef67ad9ae/pkg/cmd/roachtest/tests/tpchvec.go#L34)):*
<details><summary><i>Previously, celiala wrote…</i></summary><blockquote>

Maybe @yuzefovich ?
</blockquote></details>

It's all good - it was no longer needed and was removed.

File: pkg/cmd/roachtest/tests/django_blocklist.go (hunk 2 above)
Comment (RichardJCai) [reply pkg/cmd/roachtest/tests/django_blocklist.go:180]: We can probably just start with empty blocklists here and see what errors we run into.

**Review by rafiss — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/69829)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @celiala)

___
*[pkg/cmd/roachtest/tests/django_blocklist.go, line 180 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/69829#-Mihq2oWILDCgJf-bX54-r1-180:-Mj_WoE05B_7uHo7-a5R:bjfsa2l) ([raw file](https://github.com/cockroachdb/cockroach/blob/70567ad7bef1df02f23b916a9d1e776ef67ad9ae/pkg/cmd/roachtest/tests/django_blocklist.go#L180)):*
<details><summary><i>Previously, RichardJCai (Richard Cai) wrote…</i></summary><blockquote>

We can probably just start with empty blocklists here and see what errors we run into.
</blockquote></details>

+1 here's the full approach we should use:
- if the v21.2 blocklist is empty, then the new v22.1 blocklist should be empty too
- if the v21.2 is not empty, then set `var toolBlocklist22_1 = toolBlocklist21_2`

**Review by celiala — COMMENTED** [review]
Implemented feedback - this is RFAL:

-  In terms of exactly _when_ to merge this, let me know if I should hold off until a specific time? Otherwise, I'll plan to merge this as soon as this looks good to reviewers. Thanks!

> +1 here's the full approach we should use:
> - if the v21.2 blocklist is empty, then the new v22.1 blocklist should be empty too
> - if the v21.2 is not empty, then set var toolBlocklist22_1 = toolBlocklist21_2

Done

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/69829#-:-MkxCsIj86vqRm9DhsaB:b-mi0wjk)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @rafiss, @RichardJCai, and @yuzefovich)

___
*[pkg/cmd/roachtest/tests/django_blocklist.go, line 180 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/69829#-Mihq2oWILDCgJf-bX54-r1-180:-MkxCeAd1tmSk9naDlde:bx04tij) ([raw file](https://github.com/cockroachdb/cockroach/blob/70567ad7bef1df02f23b916a9d1e776ef67ad9ae/pkg/cmd/roachtest/tests/django_blocklist.go#L180)):*
> +1 here's the full approach we should use:
> if the v21.2 blocklist is empty, then the new v22.1 blocklist should be empty too
> if the v21.2 is not empty, then set var toolBlocklist22_1 = toolBlocklist21_2

Done

**Review by rafiss — APPROVED** [review]
thanks!

**Comment by celiala** [comment]
TFTR!

bors r+

**Pull request merged**